tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -28,10 +31,43 @@ internal class DataStreamsContextPropagator
     /// <typeparam name="TCarrier">Type of header collection</typeparam>
     public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
         where TCarrier : IBinaryHeadersCollection
+        => Inject(context, headers, Tracer.Instance.Settings.IsDataStreamsLegacyHeadersEnabled);
+
+    // Internal for testing
+    internal void Inject<TCarrier>(PathwayContext context, TCarrier headers, bool isDataStreamsLegacyHeadersEnabled)
+        where TCarrier : IBinaryHeadersCollection
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
         headers.Add(DataStreamsPropagationHeaders.PropagationKey, PathwayContextEncoder.Encode(context));
```

Comment:
Hmm this likely needs to be removed, testing locally

Reply:
I removed this, as it _should_ be added below when checking `isDataStreamsLegacyHeadersEnabled`.
However, `isDataStreamsLegacyHeadersEnabled` was always false, so it _appeared_ to be working as intended.

I fixed `isDataStreamsLegacyHeadersEnabled` returning false in 47d92fe5218987062706febd4e6fc565b319cd1f


Removed this line in f48d1dd508539ae343eb83178bebaf8df4dbf5fe

Reviewing the produced JSON it appears accurate to my eyes now